bullet_train-fields/app/controllers/concerns/fields/super_select_support.rb

```diff
@@ -20,10 +24,35 @@ def create_model_if_new(id)
   end
 
   def create_models_if_new(ids)
+    ActiveSupport::Deprecation.warn(
+      "#create_models_if_new is deprecated. " \
+      "Use #ensure_valid_ids_or_create_model instead. See an example at https://bullettrain.co/docs/field-partials/super-select#accepting-new-entries"
+    )
     ids.map do |id|
       create_model_if_new(id) do
         yield(id)
       end
     end
   end
+
+  def ensure_valid_id_or_create_model(id_or_string, collection: [], attribute: :id)
+    return id_or_string unless id_or_string.present?
+
+    valid_id = collection.find_by(attribute => id_or_string)&.public_send(attribute)
+    return valid_id.to_s unless valid_id.nil?
+
+    string = id_or_string
+    yield(string, collection)&.id&.to_s
+  end
+
+  def ensure_valid_ids_or_create_models(ids_or_strings, collection: [], attribute: :id)
+    return ids_or_strings if ids_or_strings.nil? || ids_or_strings.empty?
+
+    valid_ids = collection.valid_attribute_values([ids_or_strings], attribute: attribute).map(&:to_s)
+    new_entries = ids_or_strings - valid_ids
+
+    valid_ids + new_entries.map { |string|
+      yield(string, collection)&.id&.to_s
+    }.compact_blank
+  end
```

Comment:
I'm not quite sure about the naming with `ensure_backing_models_on` because the method does a number of things. It validates the ids, helps create missing models, and then returns the ids for reassignment on the `params`. But I'm not sure I've got anything better to make it clearer or more elegant.

Reply:
> ```ruby
> def ensure_backing_models_on(collection, id: nil, ids: [id])
>   ids = ids.compact_blank
>   return ids if ids.empty?
> 
>   existing_ids = collection.where(id: ids).ids.map(&:to_s)
>   new_ids      = ids.without(existing_ids).filter_map { yield(collection, _1)&.id&.to_s }
>   (existing_ids + new_ids).then { id ? _1.first : _1 }
> end
> ```

Nice swing! It's so terse.

* `collection` first and then the `ids` makes the method name clearer.
* `ensure_backing_models_on` also implies you don't need to pass a block.

Point taken about ids being the only relevant attribute.

Reply:
You want to commit the new method? I'd like you to get the credit.

Reply:
@pascallaliberte thank you! I've just pushed it ❤️